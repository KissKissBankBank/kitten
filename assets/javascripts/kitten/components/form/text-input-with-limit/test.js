import React from 'react'
import renderer from 'react-test-renderer'
import { TextInputWithLimit } from './index'

describe('<TextInputWithLimit />', () => {
  let component

  describe('by default', () => {
    beforeEach(() => {
      component = renderer.create(<TextInputWithLimit />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `error` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInputWithLimit error />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `disabled` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInputWithLimit disabled />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `size` prop', () => {
    beforeEach(() => {
      component = renderer.create(<TextInputWithLimit size="tiny" />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
