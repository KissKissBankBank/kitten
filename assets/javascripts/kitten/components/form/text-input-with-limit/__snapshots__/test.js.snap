// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputWithLimit /> by default matches with snapshot 1`] = `
<div
  className="text-input-with-limit__StyledTextInputWithLimit-ufrmmb-1 OpERo"
>
  <input
    className="text-input-with-limit__FocusTextInput-ufrmmb-0 cjHeAQ text-input__StyledInput-sc-11wej6v-1 bVMChg"
    defaultValue=""
    disabled={false}
    name="text"
    onChange={[Function]}
  />
  <div
    className="text-input-with-limit__StyledCounter-ufrmmb-2 jBWfBa"
    disabled={false}
  >
    80
  </div>
</div>
`;

exports[`<TextInputWithLimit /> with \`disabled\` prop matches with snapshot 1`] = `
<div
  className="text-input-with-limit__StyledTextInputWithLimit-ufrmmb-1 OpERo"
>
  <input
    className="text-input-with-limit__FocusTextInput-ufrmmb-0 cjHeAQ text-input__StyledInput-sc-11wej6v-1 bVMChg"
    defaultValue=""
    disabled={true}
    name="text"
    onChange={[Function]}
  />
  <div
    className="text-input-with-limit__StyledCounter-ufrmmb-2 kCeqRf"
    disabled={true}
  >
    80
  </div>
</div>
`;

exports[`<TextInputWithLimit /> with \`error\` prop matches with snapshot 1`] = `
<div
  className="text-input-with-limit__StyledTextInputWithLimit-ufrmmb-1 OpERo"
>
  <input
    className="text-input-with-limit__FocusTextInput-ufrmmb-0 fzFwMa text-input__StyledInput-sc-11wej6v-1 eWAcap"
    defaultValue=""
    disabled={false}
    name="text"
    onChange={[Function]}
  />
  <div
    className="text-input-with-limit__StyledCounter-ufrmmb-2 jBWfBa"
    disabled={false}
  >
    80
  </div>
</div>
`;

exports[`<TextInputWithLimit /> with \`tiny\` prop matches with snapshot 1`] = `
<div
  className="text-input-with-limit__StyledTextInputWithLimit-ufrmmb-1 OpERo"
>
  <input
    className="text-input-with-limit__FocusTextInput-ufrmmb-0 cjHeAQ text-input__StyledInput-sc-11wej6v-1 iVhmmM"
    defaultValue=""
    disabled={false}
    name="text"
    onChange={[Function]}
  />
  <div
    className="text-input-with-limit__StyledCounter-ufrmmb-2 jBWfBa"
    disabled={false}
  >
    80
  </div>
</div>
`;
